### PR TITLE
Bug Fix

### PR DESCRIPTION
**Pull request by imhassantariq**

### CODE REVIEW

File: openedx/features/edly/tasks.py [hunk 1]
```diff
@@ -39,7 +41,14 @@ def send_bulk_mail_to_students(students, param_dict, message_type):
         'comment_reply': CommentReplyNotification
     }
     message_class = message_types[message_type]
-    param_dict['site'] = Site.objects.get(id=param_dict['site_id'])
+    try:
+        # Here we don't now request is coming from LMS or CMS we need LMS Site in both cases.
```
Comment (danialmalik) [inline openedx/features/edly/tasks.py]:
```suggestion
        # Here we don't know if the request is coming from LMS or CMS but we need to use the LMS Site in both cases.
```

File: openedx/features/edly/tasks.py [hunk 2]
```diff
@@ -39,7 +41,14 @@ def send_bulk_mail_to_students(students, param_dict, message_type):
...
+        lms_base = settings.LMS_BASE
+        param_dict['site'] = Site.objects.get(Q(domain=settings.LMS_BASE) | Q(name=settings.LMS_BASE))
+    except (ObjectDoesNotExist, MultipleObjectsReturned):
+        # If accidentally we don't get the LMS site by domain name we will get the current
```
Comment (danialmalik) [inline openedx/features/edly/tasks.py]:
```suggestion
        # If we don't get the LMS site by domain name, we will get the current site
```

File: openedx/features/edly/tasks.py (hunk 1 above)
Comment (imhassantariq) [reply openedx/features/edly/tasks.py]: Done

File: openedx/features/edly/tasks.py (hunk 2 above)
Comment (imhassantariq) [reply openedx/features/edly/tasks.py]: Done

File: openedx/features/edly/tasks.py (hunk 1 above)
Comment (imhassantariq) [reply openedx/features/edly/tasks.py]: Done

File: openedx/features/edly/tasks.py (hunk 2 above)
Comment (imhassantariq) [reply openedx/features/edly/tasks.py]: Done

File: openedx/features/edly/tasks.py [hunk 3]
```diff
@@ -39,7 +41,14 @@ def send_bulk_mail_to_students(students, param_dict, message_type):
         'comment_reply': CommentReplyNotification
     }
     message_class = message_types[message_type]
-    param_dict['site'] = Site.objects.get(id=param_dict['site_id'])
+    try:
+        # Here we don't know if the request is coming from LMS or CMS but we need to use the LMS Site in both cases.
+        lms_base = settings.LMS_BASE
```
Comment (danialmalik) [inline openedx/features/edly/tasks.py]: Delete this variable if you are not using this below

**Review by danialmalik — CHANGES_REQUESTED** [review]

**Review by tehreem-sadat — APPROVED** [review]

File: openedx/features/edly/tasks.py (hunk 3 above)
Comment (imhassantariq) [reply openedx/features/edly/tasks.py]: Done

**Review by danialmalik — APPROVED** [review]

**Pull request merged**